packages/block-library/src/navigation-link/edit.js

```diff
@@ -295,6 +302,52 @@ export default function NavigationLinkEdit( {
 		[ clientId ]
 	);
 
+	// Store the colors from context as attributes for rendering
+	useEffect( () => {
+		let textColorAttr = context.textColor,
+			backgroundColorAttr = context.backgroundColor,
+			customTextColorAttr = context.customTextColor || style?.color?.text,
+			customBackgroundAttr =
+				context.customBackgroundColor || style?.color?.background;
```

Comment:
This is pretty optional, but since we're using these properties to assign in `setAttributes` it might make more sense to make this an object.

eg
```jsx
const updatedColors = {
    textColor: context.textColor,
    backgroundColor: context.backgroundColor,
    customTextColor: context.customTextColor ?? style?.color?.text,
    customBackgroundColor: context.customBackgroundColor ?? style?.color?.background
};
//...
setAttributes( updatedColors );
```

Reply:
Thanks, much cleaner!